### PR TITLE
#13916 Add big query catalog change limitation

### PR DESCRIPTION
**Pull request by Destrolaric**

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ext.bigquery/plugin.xml [hunk 1]
```diff
@@ -32,6 +32,7 @@
                         categories="bigdata">
                     <file type="jar" path="https://storage.googleapis.com/simba-bq-release/jdbc/SimbaJDBCDriverforGoogleBigQuery42_1.2.16.1020.zip"/>
                     <property name="@dbeaver-default-resultset.maxrows.sql" value="true"/>
+                    <parameter name="only-default-schema" value="true"/>
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.bigquery/plugin.xml]: `only-default-schema` &rArr; `supports-schema-change`

**Review by ShadelessFox — CHANGES_REQUESTED** [review]

**Comment by LonwoLonwo** [comment]
I do not see the ability to change SCHEMA in BigQery now. Where did you find it? Maybe you want to remove the ability to set default CATALOG?

2022-02-02 14_01_11-DBeaver Ultimate 21 3 4 - _dbeaver-db_Script-118

**Comment by Destrolaric** [comment]
@LonwoLonwo Driver doesn't support the selection of catalog, so the ability to select it is being removed due to being misleading

File: plugins/org.jkiss.dbeaver.ext.generic/plugin.xml [hunk 2]
```diff
@@ -118,26 +118,69 @@
...
+                              defaultValue="false"/>
+                    <property id="omit-single-schema" label="Omit single schema"
+                              description="Hide schema is there is only one schema on server" type="boolean"
+                              required="false" defaultValue="false"/>
+                    <property id="schema-filters-enabled" label="Use schema filters"
+                              description="Use JDBC schema filters when database doesn't support catalogs. Otherwise just read all database schemas and filter on client-side."
+                              type="boolean" required="false" defaultValue="false"/>
+                    <property id="omit-type-cache" label="Omit type cache"
+                              description="Do not use data types provided by driver" type="boolean" required="false"
+                              defaultValue="false"/>
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.generic/plugin.xml]: ?

File: plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/GenericDataSource.java [hunk 3]
```diff
@@ -251,6 +251,10 @@ protected DBPDataSourceInfo createDataSourceInfo(DBRProgressMonitor monitor, @No
         if (supportsStructCacheParam != null) {
             this.supportsStructCache = CommonUtils.toBoolean(supportsStructCacheParam);
         }
+        final Object supportsCatalogSelection = getContainer().getDriver().getDriverParameter(GenericConstants.PARAM_SUPPORTS_CATALOG_SELECTION);
+        if (supportsCatalogSelection != null) {
+            info.supportsCatalogSelection = CommonUtils.toBoolean(supportsCatalogSelection);
+        }
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/GenericDataSource.java:257]: If parameter is not set, `supportsCatalogSelection` will be `false`.

File: plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/GenericExecutionContext.java [hunk 4]
```diff
@@ -217,10 +217,10 @@ public boolean supportsCatalogChange() {
         GenericDataSource dataSource = getDataSource();
         if (dataSource.isSelectedEntityFromAPI() || !CommonUtils.isEmpty(dataSource.getQuerySetActiveDB())) {
             if (CommonUtils.isEmpty(dataSource.getSelectedEntityType())) {
-                return dataSource.hasCatalogs();
+                return dataSource.hasCatalogs() && !((GenericDataSourceInfo) dataSource.getInfo()).supportsCatalogSelection();
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/GenericExecutionContext.java]: Either extract this to a variable, or update `GenericDataSource#getInfo()` to return `GenericDataSourceInfo`.

**Review by ShadelessFox — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.ext.generic/plugin.xml [hunk 5]
```diff
@@ -118,6 +118,7 @@
 
             <driver-properties>
                 <propertyGroup label="Parameters" description="Custom driver parameters">
+                    <property id="supports-catalog-selection" label="Driver can change default catalog" description="Driver does support catalog change" type="boolean" required="false" defaultValue="true"/>;
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.generic/plugin.xml:121]: Identifier doesn't match the provided name and description

File: plugins/org.jkiss.dbeaver.ext.bigquery/plugin.xml [hunk 6]
```diff
@@ -32,6 +32,7 @@
                         categories="bigdata">
                     <file type="jar" path="https://storage.googleapis.com/simba-bq-release/jdbc/SimbaJDBCDriverforGoogleBigQuery42_1.2.16.1020.zip"/>
                     <property name="@dbeaver-default-resultset.maxrows.sql" value="true"/>
+                    <parameter name="supports-catalog-selection" value="false"/>
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.bigquery/plugin.xml:35]: `supports-catalog-selection` &rArr; `supports-catalog-change`

**Review by ShadelessFox — CHANGES_REQUESTED** [review]

**Pull request merged**